### PR TITLE
feat(web): add migration generator for webpack 5

### PR DESCRIPTION
**Pull request by jaysoo**
This PR adds a `webpack5` generator to Node and Web so users can opt into Webpack 5 for their workspace.

## Changes

- New `@nrwl/node:webpack5` and `@nrwl/web:webpack5` generators to add required  pacakges. Also includes `--revert` flag to undo the package additions.
- Added `Webpack 5 Migration` guides for Angular, React, and Node. Each has its own information relevant to that vertical.
  - https://nx-dev-git-fork-jaysoo-feat-webpack-5-migration-nrwl.vercel.app/preview/react/guides/webpack-5
- Added new `IndexHtmlWebpackPlugin` that is compatible with webpack 5.

### CODE REVIEW

File: docs/angular/api-node/generators/migrate-to-webpack-5.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+```bash
+nx generate migrate-to-webpack-5 ...
+```
+
+```bash
+nx g webpack5 ... # same
+```
+
+By default, Nx will search for `migrate-to-webpack-5` in the default collection provisioned in `angular.json`.
```
Comment (kirjai) [inline docs/angular/api-node/generators/migrate-to-webpack-5.md:15]: should this say `angular.json` or `workspace.json`, considering that this guide is for node? 

same question for `api-web`

File: docs/map.json [hunk 2]
```diff
@@ -648,6 +653,11 @@
             "id": "library",
             "file": "angular/api-node/generators/library"
           },
+          {
+            "name": "migrate to webpack 5 generator",
```
Comment (kirjai) [inline docs/map.json:657]: should the name be capitalized? same for line 359 above

**Comment by kirjai** [comment]
should we add a redirect for the deleted `nextjs-webpack5.md` page to redirect to the new webpack 5 guide?

**Comment by jaysoo** [comment]
> should we add a redirect for the deleted `nextjs-webpack5.md` page to redirect to the new webpack 5 guide?

We're trying to get away from adding redirect rules between doc changes. Users should use the search function if they don't land on the right page.

File: docs/map.json (hunk 2 above)
Comment (jaysoo) [reply docs/map.json:657]: Didn't see where the capitalized name was, but it was an older version? I kept the casing consistent with the other docs in the list.

**Pull request merged**